core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/helpers/impl/LinkHelperImpl.java

```diff
@@ -164,6 +164,10 @@ public LinkStatus validateLink(Link link, ResourceResolver resourceResolver) {
             LOG.trace("Start validation of the external link {}", link.getHref());
             link.setStatus(validateExternalLink(link.getHref()));
             LOG.trace("Completed validation of the external link {}", link.getHref());
+        } else if (link.getType() == Link.Type.CUSTOM){
+            LOG.trace("Start validation of the custom({}) link {}", link.getCustomLinkTypeProvider().getName(), link.getHref());
+            link.setStatus(link.getCustomLinkTypeProvider().validate(link.getHref()));
```

Comment:
I think we can encapsulate the `getStatus` routine into the `Link` itself. Because now the logic is a little bit strange:
We have a Link
It contains a reference to a link provider
But it does not contain status itself
For the link to retrieve a status, we "unpack" a link provider from the link. Get status. Set status to link. Then forget about the status provider again)) 

It is as if we had entity "Dog". It would not have "name" but it would have "nameProvider". And for every Dog we would go "dog.setName(dog.getNameProvider().getName())